### ISSUE TITLE
Ionic No target specified for emulator. Deploying to iPhone-6 simulator

### ISSUE THREAD

**Issue by yakupme** [driftyco/ionic-cli#317]
** BUILD SUCCEEDED **

No target specified for emulator. Deploying to iPhone-6 simulator
2015-03-19 21:01:37.299 ios-sim[4873:507] stderrPath: /Users/ezgim/firstApp/platforms/ios/cordova/console.log
2015-03-19 21:01:37.300 ios-sim[4873:507] stdoutPath: /Users/ezgim/firstApp/platforms/ios/cordova/console.log

Can someone help me to run my firstApp please . I have got cordova 4.3.0 v installed and updated ios deploy but it still does not work http://cordova.apache.org/announcements/2015/02/25/cordova-ios-3.8.0.html 

Thanks

**Comment by FrancescoMussi** [comment]
I am having the exact same issue

**Comment by yakupme** [comment]
Hi Franseco, 

After searching around two days, Finally I have found a solution for that issue. 

Create a file with name emulate.sh in your app directory. Copy the following code in your emulate.sh file.

#!/bin/bash
# Manually emulate ionic/cordova application
# Miroslav Masat 2014 
 
echo "Emulating..."
cd ./platforms/ios/build/emulator
var=$(pwd)
 
ios-sim launch "$var"/*.app


While running your project instead of running the the following command

$ ionic emulate ios

Use only ./emulate.sh
ezgims-mbp:myApp ezgim$  ./emulate.sh


Hope that will help you as well.

**Comment by mhartington** [comment]
This is something that came with cordova's cli update. Threw me off as well. 

If you need to deploy it to your device, you have to specify `--device` in the run command.

```
$ ionic platform add ios
$ ionic run ios --device
```

There were lots of changes to cordova in the latest update and this was one of them.

**Comment by yakupme** [comment]
Thanks mike. But, I have already tried the following commands. it did not work for me; 
$ ionic platform add ios
$ ionic run ios --device

By the way I became  a big fan of ionic. Keep working on this framework please :)

Its absolutely fantastic framework...

**Comment by wzomer** [comment]
+1 yakupme

**Comment by mhartington** [comment]
Can you provide the output from `ionic info`
It maybe that you need to update cordova, xcode, and ios-deploy.

**Comment by 123Tax** [comment]
Hi, I have the same issue as the op.

My Ionic info dump is:

OS: Mac OS X Yosemite
Node Version: v0.12.0
Cordova CLI: 4.3.0
Ionic Version: 1.0.0-rc.1
Ionic CLI Version: 1.3.18
Xcode version: Xcode 6.2 Build version 6C131e 
ios-sim version: 3.1.1 
ios-deploy version: 1.4.0 

Does this help mhartington?

**Comment by 123Tax** [comment]
After trying many options, I reset my mac to factory and started again (desperation lol).

I now have the same issue, my ionic info is now:
OS: Mac OS X Yosemite
Node Version: v0.12.2
Cordova CLI: 4.3.0
Ionic Version: 1.0.0-rc.2
Ionic CLI Version: 1.3.18
Xcode version: Xcode 6.2 Build version 6C131e 
ios-sim version: 3.1.1 
ios-deploy version: 1.4.0 

I am waiting to be approved by apple developer program right now, so I can only use swift :(

**Comment by mhartington** [comment]
Here is my environment info

```bash
Your system information:

OS: Mac OS X Yosemite
Node Version: v0.10.38
Cordova CLI: 4.3.0
Ionic Version: 1.0.0-rc.2
Ionic CLI Version: 1.3.19
Xcode version: Xcode 6.3 Build version 6D570
ios-sim version: 3.1.1
ios-deploy version: 1.4.0
```

So try upgrading xcode to the latest version.

**Comment by 123Tax** [comment]
Thanks, but still the same issue :( 

OS: Mac OS X Yosemite
Node Version: v0.12.2
Cordova CLI: 4.3.0
Ionic Version: 1.0.0-rc.2
Ionic CLI Version: 1.3.19
Xcode version: Xcode 6.3 Build version 6D570 
ios-sim version: 3.1.1 
ios-deploy version: 1.5.0

**Comment by jackyon** [comment]
hm... same issue happened to me...

after I tried yakupme's option, I got:

$ sudo ./emulate.sh 
sudo: ./emulate.sh: command not found


if I tried mhartington's option I got:

=== BUILD TARGET myApp OF PROJECT myApp WITH CONFIGURATION Debug ===

Check dependencies
Code Sign error: No code signing identities found: No valid signing identities (i.e. certificate and private key pair) matching the team ID “(null)” were found.
CodeSign error: code signing is required for product type 'Application' in SDK 'iOS 8.3'

** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
Error code 65 for command: xcodebuild with args: -xcconfig,/Users/jackyon/myApp/platforms/ios/cordova/build-debug.xcconfig,-project,myApp.xcodeproj,ARCHS=armv7 armv7s arm64,-target,myApp,-configuration,Debug,-sdk,iphoneos,build,VALID_ARCHS=armv7 armv7s arm64,CONFIGURATION_BUILD_DIR=/Users/jackyon/myApp/platforms/ios/build/device,SHARED_PRECOMPS_DIR=/Users/jackyon/myApp/platforms/ios/build/sharedpch
ERROR running one or more of the platforms: Error: /Users/jackyon/myApp/platforms/ios/cordova/run: Command failed with exit code 2
You may not have the required environment or OS to run this project



so sad, I can't start the project with ionic framework in my first step....

plz help! thx!

**Comment by jackyon** [comment]
one small issue confuse me, while I'm using the emulate.sh, it always opened the iphone4 Simulator, how can I switch to other iphone devices?

plz advise, thx.

**Comment by johnparn** [comment]
I've just installed ionic and I used this syntax to select the device to emulate:

```
ionic emulate ios --target="iPhone-4s"
```

To find out available emulations I run this:

```
ios-sim showdevicetypes
```

This command will return a list something like this:

```
iPhone-4s, 8.4
iPhone-5, 8.4
iPhone-5s, 8.4
iPhone-6-Plus, 8.4
iPhone-6, 8.4
iPad-2, 8.4
iPad-Retina, 8.4
iPad-Air, 8.4
Resizable-iPhone, 8.4
Resizable-iPad, 8.4
```

**Comment by rugved** [comment]
Hi john

I have same error, when I try your command I get below error. Any idea ?

sudo ionic emulate ios --target="iphone-4s"

iPhone-4s, 8.4 is not a valid target for emulator
Error: /Users/dwitibisaria/myApp/platforms/ios/cordova/run: Command failed with exit code 2
    at ChildProcess.whenDone (/usr/local/lib/node_modules/cordova/node_modules/cordova-lib/src/cordova/superspawn.js:134:23)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1015:16)
    at Process.ChildProcess._handle.onexit (child_process.js:1087:5)

**Comment by johnparn** [comment]
The only reason I can spot is that you didn't use upper case P in the
target string. Ie, try "iPhone-4s" instead of  "iphone-4s" as target.

Hope this helps!


Den 15 jul 2015 07:45 skrev "rugved" <notifications@github.com>:

**Comment by han4wluc** [comment]
I have 2 versions of the iphone on my XCode.
iPhone-6, 8.2
iPhone-6, 8.4
The problem is that I am unable to launch the 8.2 version.
`ionic emulate ios --target="iPhone-6, 8.2"`, gives error: iphone-6, 6.2 is not a valid target for emulator.
While `ionic emulate ios --target="iPhone-6"` will launch the 6.4 version.

How do I specify ionic emulate to launch with iPhone-6, 8.2?

**Comment by mohamed-ahmed** [comment]
@han4wluc What did you change the name of the devices to?

**Comment by han4wluc** [comment]
@mohamed-ahmed
If I remember correctly, 
if you run `ionic emulate ios --target="iPhone-6, 8.2"` , then change the name to `iPhone-6, 8.2`

**Comment by mohamed-ahmed** [comment]
Change the name of *what* to iPhone-6, 8.2. I think I might be missing some steps.

**Comment by han4wluc** [comment]
Change the name of your devices/simulators in XCode

XCode, => Window => Devices => and then changing the name of the devices
you should see a list of your installed simulators.

**Comment by mohamed-ahmed** [comment]
OK, I see that list, but I should change the name *from what* and *to what* ?

**Comment by han4wluc** [comment]
<img width="319" alt="screen shot 2015-09-19 at 10 10 43" src="https://cloud.githubusercontent.com/assets/8134740/9974171/c87cacb2-5eb6-11e5-988c-92d8c45640f2.png">


<img width="206" alt="screen shot 2015-09-19 at 10 15 23" src="https://cloud.githubusercontent.com/assets/8134740/9974198/63de3414-5eb7-11e5-8222-14ba343a7cc0.png">

As you see in the picture there are two `iPhone 6 Plus`, one is `8.4`, and one is `8.2`
if you run `ionic emulate ios --target="iPhone 6 Plus"`, it will run by default the 8.4 version.

In order to rename a device, you can click at the Gear icon on the bottom left and click `rename`: 
And then change the name of devices:
iPhone 6 Plus  --> iPhone 6 Plus, 8.2
iPhone 6 Plus  --> iPhone 6 Plus, 8.4

finally you can
`ionic emulate ios --target="iPhone 6 Plus, 8.2"`

**Comment by mohamed-ahmed** [comment]
I tried those steps and no matter what I type in for the target (even copied and pasted the device name from the list in Xcode to make extra sure of no spelling errors) I still get "iPhone 6 Plus is not a valid target for emulator" for whatever target name I use.

**Comment by willSonic** [comment]
I also tried all the above steps and  like mohamed-ahmed, I am unable to get it working get the message "iPhone 6 Plus is not a valid target for emulator" for whatever target name I use.

**Comment by kylefox** [comment]
I found this in `platforms/ios/cordova/lib/run.js`:

```javascript
// validate target device for ios-sim
// Valid values for "--target" (case sensitive):
var validTargets = ['iPhone-4s', 'iPhone-5', 'iPhone-5s', 'iPhone-6-Plus', 'iPhone-6',
    'iPad-2', 'iPad-Retina', 'iPad-Air', 'Resizable-iPhone', 'Resizable-iPad'];
if (args.target && validTargets.indexOf(args.target) < 0 ) {
    return Q.reject(args.target + ' is not a valid target for emulator');
}
```

It appears Cordova ignores what targets are available to `ios-sim` and limits the list to this hard-coded array. If you comment out this chunk of code you can run any of the devices listed by `ios-sim showdevicetypes`, for example: `ionic emulate ios --target="iPhone-6, 8.4"`

Would be really nice if Cordova delegated target validation to `ios-sim` rather than maintain a hard-coded list itself. 

(Maybe this fixes your issue @willSonic and @mohamed-ahmed?)

**Comment by zhouhao27** [comment]
ionic emulate ios --target="iPhone-6, 8.4" doesn't work for me. I always got the following message:

> --devicetypeid was not specified, using first available device: Apple Watch - 38mm.

It's always trying to load Apple Watch in my case no matter what I put inside the target. Any hint? Thanks

**Comment by xlozinguez** [comment]
@zhouhao27 you probably don't have a device without a paired watch in your instrument list...

**Comment by jayrmotta** [comment]
For me the only thing that worked was to rename the simulators I didn't want and keep the one I want as "iPhone 5s" for example and reinstalling it because it was not working by default. Hope it helps someone :)

**Comment by VGerris** [comment]
Just thought I should add that the recent version of cordova works correctly with the output of:
ios-sim showdevicetypes
so for example:
ionic emulate ios --target="iPhone-5, 9.2"
does what is expected when it is on the output of the previous command.
That is on 10.11.2 of OS X and Xcode 7.2, Cordova 3.9.2.
cheers

**Comment by martinjbaker** [comment]
Works for all except "iPhone-6s, 9.2" and "iPhone-6s-Plus, 9.2" for me, which is really weird.

**Comment by VGerris** [comment]
I tested those and they don't work for me either, while they are in the output of the ios-sim command.
As a workaround, you can go to the simulator and under hardware - device, choose one of those, that works for me. I get really strange large screens for the plus versions by the way :).

**Comment by martinjbaker** [comment]
Apparently a known bug which will be fixed in next update. The plus screens are far bigger because they are 3x resolution rather than 2x, so you'll probably need to zoom the simulator down in the View menu.

**Comment by matthieu-D** [comment]
@kylefox Thanks

**Comment by zhouhao27** [comment]
I remove the platforms folder and add the platform again by:

> ionic platform add ios

Solve the issue.

**Comment by VGerris** [comment]
This is just such a piece of shit. I updated everything to recent versions and now I get after:
ionic emulate ios --target="iPhone-5, 9.2"
--devicetypeid was not specified, using first available device: Apple Watch - 38mm.

So I also have the watch loading everytime now.
Everytime I try to get somewhere with this framework, something else break.
It would be great if I could just start developing instead of fixing things all the time that should not be broken.
Sorry, I had to nag about that.
Anyway, to fix that shit what worked indeed (thank @zhouhao27 ) was to remove the ios folder from the platforms folder and remove and re-add the ios platform plugin: cordova run ios

Then I had all kinds of issues with older plugins, but I'll not go on about that.
The command:
ionic emulate ios --target="iPhone-5, 9.2"
worked again after it to run the specific emulator.
